src/Scan/BasePolling.spec.ts

```diff
@@ -0,0 +1,221 @@
+      // arrange
+      const options = {
+        scanId: 'scanId',
+        interval: 5000
+      };
+
+      // act
+      new BasePolling(
+        options,
+        instance(scanManagerMock),
+        instance(breakpointMock)
+      );
+
+      // assert
+      verify(loggerSpy.warn(`Warning: polling interval is too small.`)).once();
+      verify(
+        loggerSpy.warn(`The recommended way to set polling interval to 10s.`)
+      ).once();
+    });
+  });
+
+  describe('start', () => {
+    it('should start polling and stop on breakpoint exception', async () => {
+      // arrange
+      const scanId = 'scanId';
+      const sut = new BasePolling(
+        {
+          scanId,
+          interval: 1
+        },
+        instance(scanManagerMock),
+        instance(breakpointMock)
+      );
+      const response: ScanState = {
+        numberOfHighSeverityIssues: 0,
+        numberOfCriticalSeverityIssues: 0,
+        numberOfLowSeverityIssues: 0,
+        numberOfMediumSeverityIssues: 0,
+        status: ScanStatus.RUNNING
+      };
+      when(scanManagerMock.status(scanId))
+        .thenResolve(response)
+        .thenResolve({ ...response, numberOfLowSeverityIssues: 1 });
+
+      when(
+        breakpointMock.execute(
+          objectContaining({ numberOfLowSeverityIssues: 1 })
+        )
+      ).thenReject(new Error('breakpoint error'));
+      // act
+      const act = sut.start();
+      // assert
+      await expect(act).rejects.toThrow('breakpoint error');
+      verify(loggerSpy.log('Starting polling...')).once();
+      verify(scanManagerMock.status(scanId)).twice();
+    });
+
+    it.each([
+      ScanStatus.DONE,
+      ScanStatus.DISRUPTED,
+      ScanStatus.FAILED,
+      ScanStatus.STOPPED
+    ])(
+      'should start polling and stop on scan status changed to "%s"',
+      async (status) => {
+        // arrange
+        const scanId = 'scanId';
+        const sut = new BasePolling(
+          {
+            scanId,
+            interval: 1
+          },
+          instance(scanManagerMock),
+          instance(breakpointMock)
+        );
+        const response: ScanState = {
+          numberOfHighSeverityIssues: 0,
+          numberOfCriticalSeverityIssues: 0,
+          numberOfLowSeverityIssues: 0,
+          numberOfMediumSeverityIssues: 0,
+          status: ScanStatus.RUNNING
+        };
+        when(scanManagerMock.status(scanId))
+          .thenResolve(response)
+          .thenResolve({ ...response, status });
+
+        // act
+        await sut.start();
+        // assert
+        verify(scanManagerMock.status(scanId)).twice();
+      }
+    );
+
+    it('should start polling and stop on timeout', async () => {
+      // arrange
+      const scanId = 'scanId';
+      const timeout = 10000;
+      const sut = new BasePolling(
+        {
+          scanId,
+          timeout,
+          interval: 1
+        },
+        instance(scanManagerMock),
+        instance(breakpointMock)
+      );
```

Comment:
Create a SUT on the `beforeEach` hook; utilize the spy on the options to make modifications when necessary.